docs/platform/connector-development/connector-builder-ui/authentication.md

```diff
@@ -214,26 +296,84 @@ We will walk through each part of the configuration below. Throughout this, we w
-  - Note: this **does _not_ support dynamic expiration durations of session tokens**. If your token expiration duration is dynamic, you should set the `Expiration Duration` field to the expected minimum duration to avoid problems during syncing.
+  - **Note**: This **does not support dynamic expiration durations of session tokens**. If your token expiration duration is dynamic, you should set the `Expiration Duration` field to the expected minimum duration to avoid problems during syncing.
   - For Metabase, the token retrieved from the `/api/session` endpoint expires after 14 days by default, so this value can be set to `P2W` or `P14D`.
-- `Data Request Authentication` - configures how the session token is used to authenticate the data requests made to the API
+- `Data Request Authentication` - Configures how the session token is used to authenticate the data requests made to the API
   - Choose `API Key` if your session token needs to be injected into a query parameter or header of the data requests.
     - Metabase takes in the session token through a specific header, so this would be set to `API Key`, Inject Session Token into outgoing HTTP Request would be set to `Header`, and Header Name would be set to `X-Metabase-Session`.
   - Choose `Bearer` if your session token needs to be sent as a standard Bearer token.
 
+## Selective Authenticator
+
+Selective Authenticator allows you to dynamically choose between different authentication methods based on a configuration property value. This is useful when your connector needs to support multiple authentication options and let users select which one to use.
+
+### Configuration
+
+- **Authenticator Selection Path**: An array of strings defining the path in the configuration to the property that determines which authenticator to use
+- **Authenticators**: A mapping of authenticator names to their respective configurations
+
+### How it works
+
+The Selective Authenticator evaluates the value at the specified configuration path and uses it to select the appropriate authenticator from the available options. For example, if the selection path points to a config property with value "api_key", it will use the authenticator mapped to "api_key".
+
+### Example
+
+If you want to support both API Key and OAuth authentication:
+
+1. **Authenticator Selection Path**: `["auth", "type"]`
+2. **Authenticators**: 
+   - `api_key`: Configure an API Key authenticator
+   - `oauth`: Configure an OAuth authenticator
+
+When users configure their connector, they would set `auth.type` to either "api_key" or "oauth", and the Selective Authenticator will automatically use the corresponding authentication method.
+
+### Use Cases
+
+- APIs that support multiple authentication methods
+- Connectors that need to work with different API versions requiring different auth
+- Providing flexibility for users to choose their preferred authentication method
+
```

Comment:
This isn't really well-supported by the UI today, because we don't support nested inputs in the Inputs tab. Users can only really use this if they switch to YAML mode where they can fully edit the connector spec.

I'd say let's just axe this section for now, or add a note saying that users will need to switch to YAML mode to configure the spec with nested inputs